### ISSUE TITLE
Unable to use GZoltar on Command-Line

### ISSUE THREAD

**Issue by riteshreddy99** [GZoltar/gzoltar#52]
### Context
*Hi, @jose,
I tried running the tool on terminal on Ubuntu and it is not able run the command` mvn package` .*

### Steps to Reproduce
Ran `mvn package ` command to start GZoltar

### GZoltar arguments

Command-Line Interface*

### Environment (please complete the following information, if relevant):
 -Distributor ID:	Ubuntu
-Description:	Ubuntu 20.04.4 LTS
-Release:	20.04
openjdk 11.0.14.1 2022-02-08
OpenJDK Runtime Environment (build 11.0.14.1+1-Ubuntu-0ubuntu1.20.04)
OpenJDK 64-Bit Server VM (build 11.0.14.1+1-Ubuntu-0ubuntu1.20.04, mixed mode, shari

Thanks! Would appreciate an early respons
![gzoltar](https://user-images.githubusercontent.com/32620436/162805155-c285c1e2-9862-4091-b865-75e0a399693a.PNG)
e

**Comment by jose** [comment]
Hi @riteshreddy99,

I'm confused.  Can't you compile GZoltar ou can't you run GZoltar from the command line?

- From your instructions and screenshot it seems you're having problems compiling it because some GZoltar's tests are failing.  It might be they are indeed failing due to a fault or it might be they are simplying failing because you are not using the recommend Java version (i.e., JDK 8).  I would recommend you compile GZoltar with `mvn clean package -DskipTests`.

- From this issue's title it seems you're having problems running GZoltar on the command line but I failed to see how, as the instructions point to the compilation of GZoltar.


--
Best,
Jose

**Comment by riteshreddy99** [comment]
Yeah that was the issue, I was not using JDK 8. After switching to JDK 8 the build was successful. Could you please help me out on how I can use the GZoltar VSCode extension on these two datasets containing java projects.

https://github.com/jkoppel/QuixBugs

https://github.com/Spirals-Team/IntroClassJava

Thanks.
Ritesh

**Comment by jose** [comment]
- Projects on [QuixBugs](https://github.com/jkoppel/QuixBugs) do not follow the standard structure used in Java/Maven projects and therefore you will have a very hard time just importing them to VSCode.

- Project on [IntroClassJava](https://github.com/Spirals-Team/IntroClassJava), on the other hand, seem to follow the standard structure used in Java/Maven projects.  I haven't tried VSCode + GZoltar + [IntroClassJava](https://github.com/Spirals-Team/IntroClassJava), but I would say you would have to:
1. Open VSCode.
2. Import a single project, e.g., https://github.com/Spirals-Team/IntroClassJava/tree/master/dataset/checksum/08c7ea4ac39aa6a5ab206393bb4412de9d2c365ecdda9c1b391be963c1811014ed23d2722d7433b8e8a95305eee314d39da4950f31e01f9147f90af91a5c433a/006.
3. Run GZoltar as described in [here](https://github.com/GZoltar/gzoltar/tree/master/com.gzoltar.vscode#usage).


--
Best,
Jose

**Comment by riteshreddy99** [comment]
Is there any other dataset I could use to test the tool on other than the https://github.com/jkoppel/QuixBugs

**Comment by jose** [comment]
Yes, there are many others you could use, e.g.: [Bugs.jar](https://github.com/bugs-dot-jar/bugs-dot-jar), [Bears](https://github.com/bears-bugs/bears-benchmark), and of course the well known [Defects4J](https://github.com/rjust/defects4j).

**Comment by riteshreddy99** [comment]
Thanks! I will check it out.

**Comment by riteshreddy99** [comment]
For some reason GZoltar is unable to detect the open folder of the IntroClassJava project. What could be the issue?


![Gzoltar_vscode](https://user-images.githubusercontent.com/32620436/163687496-a3e180f9-f8f5-4203-82a7-afbb21ca999f.PNG)

**Comment by riteshreddy99** [comment]
Hey @jose,
 This is the error I'm facing upon trying to test the example GZoltar maven projects. Please take a look.
Thanks,
Ritesh
![gzoltar-error](https://user-images.githubusercontent.com/32620436/163707417-83ee96b2-477f-457a-a487-a161123b0d59.PNG)

**Comment by jose** [comment]
> For some reason GZoltar is unable to detect the open folder of the IntroClassJava project. What could be the issue?

By just looking at the screenshot it is impossible to guess what the issue could be as I don't see any error message of any kind.

> This is the error I'm facing upon trying to test the example GZoltar maven projects. Please take a look.

It seens your IDE is complaining of a missing **runtime** dependency.  Have you managed to successfully compile/package GZoltar?


--
Best,
Jose

**Comment by riteshreddy99** [comment]
I just installed the GZoltar Extension from the VSCode Marketplace.

**Comment by jose** [comment]
And that's fine, but the project you're using (i.e., [maven-examples](https://github.com/GZoltar/gzoltar/tree/master/com.gzoltar.maven.examples)) requires you to successfully compile GZoltar first.


--
Best,
Jose

**Comment by riteshreddy99** [comment]
```
mvn clean test-compile
mvn -P sufire gzoltar:prepare-agent test
mvn gzoltar:fl-report
```
This is what is supposed to be run?

**Comment by jose** [comment]
> This is what is supposed to be run?

No.  As described in [this README file](https://github.com/GZoltar/gzoltar/tree/master/com.gzoltar.maven.examples#maven-example), those are the commands to run the [maven-examples](https://github.com/GZoltar/gzoltar/tree/master/com.gzoltar.maven.examples) from the command line using "Online instrumentation with maven-sufire test case executor". 

That same [README file](https://github.com/GZoltar/gzoltar/tree/master/com.gzoltar.maven.examples#maven-example) also says:

> Note that the GZoltar project must be 'packaged' before running any of the usage examples described below, i.e., mvn package must be executed on the root directory.

which is exactly what I was trying to tell you.

**Comment by riteshreddy99** [comment]
Okay, so I ran `mvn package`

Upon, trying to open vulnerability report an error pops up.

![gzoltar-maven](https://user-images.githubusercontent.com/32620436/163710348-f168ea83-279b-4aea-822f-44f708fc3a37.PNG)
